independent-projects/tools/common/src/main/java/io/quarkus/devtools/project/extensions/ExtensionManager.java

```diff
@@ -0,0 +1,109 @@
+     *
+     * @return The list of {@link AppArtifactCoords} installed in the project build file(s).
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    List<AppArtifactCoords> getInstalled() throws IOException;
+
+    /**
+     * Read build file(s) to check if an extension is installed in this Quarkus project.
+     *
+     * @param key the {@link AppArtifactKey} of the extension to check
+     * @return true if it's installed
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    default boolean isInstalled(AppArtifactKey key) throws IOException {
+        return getInstalled().stream().anyMatch(i -> Objects.equals(i.getKey(), key));
+    }
+
+    /**
+     * Check that the Quarkus Platform bom is defined in the project.
+     * The Quarkus Platform Bom role is to define version to use for Quarkus extensions.
+     *
+     * @return true if it's defined, false else
+     * @throws IOException if a problem occurs while reading the project build file(s)
+     */
+    boolean hasQuarkusPlatformBom() throws IOException;
+
+    /**
+     * This is going to install/add all the specified extensions to the project build file(s).
+     *
+     * <pre>
+     *   - If the project Quarkus platform bom is not defined, an {@link IllegalStateException} will be thrown
+     *   - Extensions which are already installed will ALWAYS be skipped whatever the specified version
+     *   - The provided version will be used wasn't already installed
+     * </pre>
+     *
+     * @param coords the list of {@link AppArtifactCoords} for the extensions to install
+     * @return the {@link InstallResult}
+     * @throws IOException if a problem occurs while reading/writing the project build file(s)
+     */
+    InstallResult install(List<AppArtifactCoords> coords) throws IOException;
+
+    /**
+     * This is going to uninstall/remove all the specified extensions from the project build file(s).
+     *
+     * This is ignoring the {@link Extension} version
+     *
+     * @param keys the set of {@link AppArtifactKey} for the extensions to uninstall
+     * @return the {@link InstallResult}
+     * @throws IOException if a problem occurs while reading/writing the project build file(s)
+     */
+    UninstallResult uninstall(Set<AppArtifactKey> keys) throws IOException;
+
+    class InstallResult {
+        private final int installed;
+
+        public InstallResult(int installed) {
+            this.installed = installed;
+        }
+
+        public int getInstalled() {
+            return installed;
+        }
+
+        public boolean isSourceUpdated() {
+            return installed > 0;
+        }
+    }
+
+    class UninstallResult {
+        private final int uninstalled;
+
+        public UninstallResult(int uninstalled) {
+            this.uninstalled = uninstalled;
+        }
+
+        public int getUninstalled() {
```

Comment:
Instead of returning how many extensions were installed/uninstalled wouldn't be better to return *which* extensions were installed/uninstalled (the collection size would return the total of installed/uninstalled items)?

Reply:
@gastaldi that's a good idea! and it could be done without changing the contract. Let's do it in another PR.

Reply:
yeah and I need to print it too, it's not printed anymore when installing/uninstalling.. i'll create an issue as soon as we merge that one.